apps/condo/domains/miniapp/schema/SendAppPushMessageService.js

```diff
@@ -87,16 +123,55 @@ const SendAppPushMessageService = new GQLCustomSchema('SendAppPushMessageService
         {
             access: access.canSendAppPushMessage,
             schema: 'sendAppPushMessage(data: SendAppPushMessageInput!): SendAppPushMessageOutput',
-            resolver: async (parent, args, context, info, extra = {}) => {
+            resolver: async (parent, args, context) => {
                 const { data: argsData } = args
                 const { dv, sender, app, user, type, data, uniqKey } = argsData
-                const { B2CAppContext } = data
+                const { B2CAppContext, resident, property, unitType, unitName } = data
+                const residentId = get(resident, 'id')
+                const propertyId = get(property, 'id')
+                let residentIdFromProperty = null
 
                 checkDvAndSender(argsData, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
 
                 const userExisted = await User.getOne(context, { id: user.id, deletedAt: null })
 
-                if (!userExisted)  throw new GQLError(ERRORS.USER_NOT_FOUND, context)
+                if (!userExisted) throw new GQLError(ERRORS.USER_NOT_FOUND, context)
+                if (!residentId && !(propertyId && unitType && unitName)) throw new GQLError(ERRORS.RESIDENT_OR_PROPERTY_ID_MISSING, context)
+
+                /**
+                 * Mobile APP needs either resident id or property id + unitType/unitName to detect resident
+                 *
+                 * So we require here either of these fields to be provided and check validity/existence of provided id's
+                 * */
+                if (residentId) {
+                    /** resident must belong to the user */
+                    const residentWhere = {
+                        id: residentId,
+                        user: { id: user.id },
+                        deletedAt: null,
+                    }
+                    const residentExisted = await Resident.getOne(context, residentWhere)
+
+                    if (!residentExisted) throw new GQLError(ERRORS.RESIDENT_NOT_FOUND, context)
+                } else {
```

Comment:
It looks like a security issue. Are you trying to fake an actual resident and bypass the access?

Reply:
not at all. miniapp backend sometimes would have resident id and other times property id and unitType/unitName, so I check here that it resolved to the resident, connected to the user.